### PR TITLE
2048: Let user decide if he wants to continue the game

### PR DESCRIPTION
**Pull request by foragerDev**
Before game finish when a player has reached target tile,
Now player will be able to decide if he wants to continue or not.

### CODE REVIEW

File: Userland/Games/2048/Game.h [hunk 1]
```diff
@@ -119,6 +119,7 @@ class Game final {
     u32 m_target_tile { 0 };
 
     bool m_evil_ai { false };
+    bool m_want_to_continue { false };
```
Comment (alimpfard) [inline Userland/Games/2048/Game.h:122]: This flag isn't reset even after restarting a game, so there won't be any more "winning" until the user relaunches the game.

File: Userland/Games/2048/main.cpp [hunk 2]
```diff
@@ -151,14 +151,18 @@ int main(int argc, char** argv)
...
             break;
-        case Game::MoveOutcome::Won:
+        case Game::MoveOutcome::Won: {
             update();
-            GUI::MessageBox::show(window,
-                String::formatted("You reached {} in {} turns with a score of {}", game.target_tile(), game.turns(), game.score()),
-                "You won!",
-                GUI::MessageBox::Type::Information);
-            start_a_new_game();
+            auto message_box = GUI::MessageBox::construct(window, "Congratulations! You won the game, Do you still want to continue?",
```
Comment (alimpfard) [inline Userland/Games/2048/main.cpp:156]: The "<score> in <yay many moves>" message doesn't exist anymore, that doesn't seem right?

**Review by alimpfard — COMMENTED** [review]
I'm not sure if continuing past the configurable target tile is something people want to do tbh?

File: Userland/Games/2048/Game.h (hunk 1 above)
Comment (foragerDev) [reply Userland/Games/2048/Game.h:122]: Yes, I missed that corner case.

File: Userland/Games/2048/main.cpp (hunk 2 above)
Comment (foragerDev) [reply Userland/Games/2048/main.cpp:156]: Now that will get displayed when the board is stalled. But we can show user this information.

**Comment by foragerDev** [comment]
@alimpfard I play this game usually, If I reach the target time, I would like to keep playing as far as I can.

**Comment by alimpfard** [comment]
Alright, I suppose
You _could_ just set the target tile to the maximum possible tile though...

**Comment by foragerDev** [comment]
yeah, I can but sometimes, I can not predict, how long will I go, as game is 2048, so general target should 2048 and then If I wish to continue or not I should be able to choose.

**Comment by alimpfard** [comment]
That's fine I guess, just fix those two issues : )

File: Userland/Games/2048/Game.h (hunk 1 above)
Comment (foragerDev) [reply Userland/Games/2048/Game.h:122]: @alimpfard if you see the code of start `start_a_new_game`, a new game object is constructed there, so `m_want_to_continue` is automatically reset to `false`.

File: Userland/Games/2048/main.cpp (hunk 2 above)
Comment (foragerDev) [reply Userland/Games/2048/main.cpp:156]: I have fixed this issue.

File: Userland/Games/2048/Game.h (hunk 1 above)
Comment (alimpfard) [inline Userland/Games/2048/Game.h:122]: I still don't see this being reset to `false` anywhere.

Comment (foragerDev) [reply Userland/Games/2048/Game.h:122]: @alimpfard look at `start_a_new_game` when it is called, it automatically create new game object, which default construct it with false? is not it ?

Comment (alimpfard) [reply Userland/Games/2048/Game.h:122]: Yep, sorry, my bad.

Comment (foragerDev) [reply Userland/Games/2048/Game.h:122]: That's fine. :)

**Review by alimpfard — APPROVED** [review]

**Pull request merged**